lib/go-atscfg/cachedotconfig.go

```diff
@@ -67,22 +67,24 @@ func makeCacheDotConfigEdge(
 	}
 	warnings := []string{}
 
-	if server.Profile == nil {
-		return Cfg{}, makeErr(warnings, "server missing profile")
+	if server.ProfileNames == nil && len(server.ProfileNames) == 0 {
+		return Cfg{}, makeErr(warnings, "server missing profiles")
 	}
 
 	profileServerIDsMap := map[int]struct{}{}
 	for _, sv := range servers {
-		if sv.Profile == nil {
+		if sv.ProfileNames == nil && len(sv.ProfileNames) == 0 {
 			warnings = append(warnings, "servers had server with nil profile, skipping!")
 			continue
 		}
 		if sv.ID == nil {
 			warnings = append(warnings, "servers had server with nil id, skipping!")
 			continue
 		}
-		if *sv.Profile != *server.Profile {
-			continue
+		for i, _ := range server.ProfileNames {
+			if sv.ProfileNames[i] != server.ProfileNames[i] {
+				continue
+			}
```

Comment:
This doesn't replicate the logic it's replacing. The `continue` here is associated with your new `for`. So you'd need to do something like
```go
if len(sv.ProfileNames) != len(server.ProfileNames) {
    continue
}
profilesTheSame := true
for i, _ := range server.ProfileNames {
    if sv.ProfileNames[i] != server.ProfileNames[i] {
        profilesTheSame = false
        break
    }
}
if !profilesTheSame {
    continue
}
```
Note that you should also check the lengths before iterating, otherwise it could segfault.

Reply:
Hmm.. I see.